app/routines/generate_reset_code.rb

```diff
@@ -0,0 +1,18 @@
+class GenerateResetCode
+
+  lev_routine
+
+  protected
+
+  def exec(identity, expiration_period = ResetCode::DEFAULT_EXPIRATION_PERIOD)
+    identity.reset_code ||= identity.build_reset_code
```

Comment:
Doesn't this code need to save the `identity` for when `reset_code` has to be built?  I see tests that save the `identity` but don't see saving in the real code